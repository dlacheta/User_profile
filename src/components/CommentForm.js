import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import uuid from 'uuid';
import { addComment } from '../actions/rootActions';

export class CommentForm extends React.Component {
  constructor(props) {
    super(props);
    this.onCommentSubmit = this.onCommentSubmit.bind(this);
  }

  onCommentSubmit(event) {
    const { addCommentDispatch } = this.props;
    event.preventDefault();
    const newComment = {
      id: uuid(),
      name: 'Rachel Zein',
      date: +new Date(),
      comment: event.target.add_comment.value,
      avatar: 'user.png',
    };
    addCommentDispatch(newComment);
    event.target.add_comment.value = '';
  }

  render() {
    return (
      <div className="comment-form">
        <form onSubmit={this.onCommentSubmit}>
          <input name="add_comment" type="text" placeholder="Add a comment" />
        </form>
      </div>
    );
  }
}

CommentForm.propTypes = {
  addCommentDispatch: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  addCommentDispatch: newComment => dispatch(addComment(newComment)),
});

export default connect(
  null,
  mapDispatchToProps,
)(CommentForm);
