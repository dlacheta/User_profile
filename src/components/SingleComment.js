import React from 'react';
import PropTypes from 'prop-types';
import dateDiffInDays from '../func/dateDiffInDays';

const SingleComment = ({
  comment: {
    name, comment, date, avatar,
  },
}) => (
  <div className="comments-scroll">
    <img className="comment-img" src={avatar} alt={name} />
    <div className="container-fluid">
      <div className="comment-header">
        <h4>{name}</h4>
        <span className="comment-date">{dateDiffInDays(date)}</span>
      </div>
      {comment}
    </div>
  </div>
);

SingleComment.propTypes = {
  comment: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    date: PropTypes.number,
    comment: PropTypes.string,
  }),
};

SingleComment.defaultProps = {
  comment: {},
};

export default SingleComment;
