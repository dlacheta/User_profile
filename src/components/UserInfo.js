import React from 'react';
import { connect } from 'react-redux';
import Modal from 'react-modal';
import PropTypes from 'prop-types';
import {
  getUser,
  addLike,
  removeLike,
  addFollower,
  removeFollower,
  toogleModal,
} from '../actions/rootActions';

if (process.env.NODE_ENV !== 'test') Modal.setAppElement('#app');

export class UserInfo extends React.Component {
  componentDidMount() {
    const { getUserDispatch } = this.props;
    getUserDispatch();
  }

  render() {
    const {
      addLikeDispatch,
      addFollowerDispatch,
      shareUrlDispatch,
      removeLikeDispatch,
      removeFollowerDispatch,
      user: {
        name, city, country, likes, following, followers, modalIsOpen, liked, followed,
      },
    } = this.props;
    return (
      <div className="user-data">
        <div className="user-info-container">
          <img className="user-img" src="user.png" alt={name} />
          <i onClick={shareUrlDispatch} className="material-icons share-icon">
            exit_to_app
          </i>
          <div className="user-info">
            <h1>
              {name}
              {liked ? (
                <i onClick={removeLikeDispatch} className="material-icons like-icon" alt="Unlike">
                  favorite
                </i>
              ) : (
                <i onClick={addLikeDispatch} className="material-icons like-icon" alt="Like">
                  favorite_border
                </i>
              )}
            </h1>
            <h2>{`${city}, ${country}`}</h2>
          </div>
        </div>
        <div className="user-follow">
          <div>
            <h3>{likes}</h3>
            Likes
          </div>
          <div className="follow-divider" />
          <div>
            <h3>{following}</h3>
            Following
          </div>
          <div className="follow-divider" />
          <div>
            <h3>{followers}</h3>
            Followers
          </div>
          {followed ? (
            <button onClick={removeFollowerDispatch} type="button" className="follow-button">
              UNFOLLOW
            </button>
          ) : (
            <button onClick={addFollowerDispatch} type="button" className="follow-button">
              FOLLOW
            </button>
          )}
        </div>
        <Modal className="modal" isOpen={modalIsOpen} onRequestClose={shareUrlDispatch}>
          <i onClick={shareUrlDispatch} className="material-icons close-button">
            close
          </i>
          <h3 className="modal__title">Copy link below to share:</h3>
          {window.location.href}
        </Modal>
      </div>
    );
  }
}

UserInfo.propTypes = {
  user: PropTypes.shape({
    name: PropTypes.string,
    city: PropTypes.string,
    country: PropTypes.string,
    likes: PropTypes.number,
    following: PropTypes.number,
    followers: PropTypes.number,
  }),
  addLikeDispatch: PropTypes.func.isRequired,
  removeLikeDispatch: PropTypes.func.isRequired,
  addFollowerDispatch: PropTypes.func.isRequired,
  removeFollowerDispatch: PropTypes.func.isRequired,
  shareUrlDispatch: PropTypes.func.isRequired,
  getUserDispatch: PropTypes.func.isRequired,
};
UserInfo.defaultProps = {
  user: {},
};

const mapStateToProps = state => ({ user: state });

const mapDispatchToProps = dispatch => ({
  getUserDispatch: () => dispatch(getUser()),
  addLikeDispatch: () => dispatch(addLike()),
  removeLikeDispatch: () => dispatch(removeLike()),
  addFollowerDispatch: () => dispatch(addFollower()),
  removeFollowerDispatch: () => dispatch(removeFollower()),
  shareUrlDispatch: () => dispatch(toogleModal()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(UserInfo);
