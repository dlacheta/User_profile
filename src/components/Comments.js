import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getUser, toogleComments } from '../actions/rootActions';
import SingleComment from './SingleComment';
import CommentForm from './CommentForm';

export class Comments extends React.Component {
  constructor(props) {
    super(props);
    this.messagesEnd = React.createRef();
    this.scrollToBottom = this.scrollToBottom.bind(this);
  }

  componentDidMount() {
    const { getUserDispatch, commentsVisible } = this.props;
    getUserDispatch();
    if (commentsVisible) {
      this.scrollToBottom();
    }
  }

  componentDidUpdate() {
    const { commentsVisible } = this.props;
    if (commentsVisible) {
      this.scrollToBottom();
    }
  }

  scrollToBottom() {
    this.messagesEnd.current.scrollIntoView({ behavior: 'smooth' });
  }

  render() {
    const { allComments, commentsVisible, toogleCommentsDispatch } = this.props;
    return (
      <div className="comments">
        <button type="button" onClick={toogleCommentsDispatch}>
          {commentsVisible
            ? `Hide comments (${allComments.length})`
            : `Show comments (${allComments.length})`}
        </button>
        {commentsVisible ? (
          <div className="container-fluid">
            <div className="comments-scroll-container">
              {allComments.map(comment => (
                <SingleComment key={comment.id} comment={comment} />
              ))}
              <div ref={this.messagesEnd} />
            </div>
            <CommentForm />
          </div>
        ) : null}
      </div>
    );
  }
}

Comments.propTypes = {
  getUserDispatch: PropTypes.func.isRequired,
  allComments: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
      date: PropTypes.number,
      comment: PropTypes.string,
    }),
  ),
  commentsVisible: PropTypes.bool,
  toogleCommentsDispatch: PropTypes.func.isRequired,
};

Comments.defaultProps = {
  allComments: [],
  commentsVisible: true,
};

const mapStateToProps = state => ({
  allComments: state.comments.sort((a, b) => {
    if (a.date < b.date) return -1;
    if (a.date > b.date) return 1;
    return 0;
  }),
  commentsVisible: state.commentsVisible,
});

const mapDispatchToProps = dispatch => ({
  getUserDispatch: () => dispatch(getUser()),
  toogleCommentsDispatch: () => dispatch(toogleComments()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Comments);
