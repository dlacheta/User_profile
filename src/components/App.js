import React from 'react';
import UserInfo from './UserInfo';
import Comments from './Comments';

const App = () => (
  <div className="container">
    <div className="top-dark" />
    <div className="inside-container">
      <UserInfo />
      <Comments />
    </div>
  </div>
);

export default App;
