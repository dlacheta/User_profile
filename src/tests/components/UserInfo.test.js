import React from 'react';
import { shallow } from 'enzyme';
import { UserInfo } from '../../components/UserInfo';

it('UserInfo component renders correctly', () => {
  const addLikeDispatch = jest.fn();
  const removeLikeDispatch = jest.fn();
  const addFollowerDispatch = jest.fn();
  const removeFollowerDispatch = jest.fn();
  const shareUrlDispatch = jest.fn();
  const getUserDispatch = jest.fn();
  const wrapper = shallow(
    <UserInfo
      addLikeDispatch={addLikeDispatch}
      removeLikeDispatch={removeLikeDispatch}
      addFollowerDispatch={addFollowerDispatch}
      removeFollowerDispatch={removeFollowerDispatch}
      shareUrlDispatch={shareUrlDispatch}
      getUserDispatch={getUserDispatch}
    />,
  );
  expect(wrapper).toMatchSnapshot();
});
