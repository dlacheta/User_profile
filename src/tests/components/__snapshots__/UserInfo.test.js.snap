// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserInfo
    addFollowerDispatch={[MockFunction]}
    addLikeDispatch={[MockFunction]}
    getUserDispatch={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    removeFollowerDispatch={[MockFunction]}
    removeLikeDispatch={[MockFunction]}
    shareUrlDispatch={[MockFunction]}
    user={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="user-info-container"
        >
          <img
            className="user-img"
            src="user.png"
          />
          <i
            className="material-icons share-icon"
            onClick={[MockFunction]}
          >
            exit_to_app
          </i>
          <div
            className="user-info"
          >
            <h1>
              <i
                alt="Like"
                className="material-icons like-icon"
                onClick={[MockFunction]}
              >
                favorite_border
              </i>
            </h1>
            <h2>
              undefined, undefined
            </h2>
          </div>
        </div>,
        <div
          className="user-follow"
        >
          <div>
            <h3 />
            Likes
          </div>
          <div
            className="follow-divider"
          />
          <div>
            <h3 />
            Following
          </div>
          <div
            className="follow-divider"
          />
          <div>
            <h3 />
            Followers
          </div>
          <button
            className="follow-button"
            onClick={[MockFunction]}
            type="button"
          >
            FOLLOW
          </button>
        </div>,
        <Modal
          ariaHideApp={true}
          bodyOpenClassName="ReactModal__Body--open"
          className="modal"
          closeTimeoutMS={0}
          isOpen={false}
          onRequestClose={[MockFunction]}
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          role="dialog"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
        >
          <i
            className="material-icons close-button"
            onClick={[MockFunction]}
          >
            close
          </i>
          <h3
            className="modal__title"
          >
            Copy link below to share:
          </h3>
          http://localhost/
        </Modal>,
      ],
      "className": "user-data",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              className="user-img"
              src="user.png"
            />,
            <i
              className="material-icons share-icon"
              onClick={[MockFunction]}
            >
              exit_to_app
            </i>,
            <div
              className="user-info"
            >
              <h1>
                <i
                  alt="Like"
                  className="material-icons like-icon"
                  onClick={[MockFunction]}
                >
                  favorite_border
                </i>
              </h1>
              <h2>
                undefined, undefined
              </h2>
            </div>,
          ],
          "className": "user-info-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": undefined,
              "className": "user-img",
              "src": "user.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "exit_to_app",
              "className": "material-icons share-icon",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "exit_to_app",
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  <i
                    alt="Like"
                    className="material-icons like-icon"
                    onClick={[MockFunction]}
                  >
                    favorite_border
                  </i>
                </h1>,
                <h2>
                  undefined, undefined
                </h2>,
              ],
              "className": "user-info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    <i
                      alt="Like"
                      className="material-icons like-icon"
                      onClick={[MockFunction]}
                    >
                      favorite_border
                    </i>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "Like",
                      "children": "favorite_border",
                      "className": "material-icons like-icon",
                      "onClick": [MockFunction],
                    },
                    "ref": null,
                    "rendered": "favorite_border",
                    "type": "i",
                  },
                ],
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "undefined, undefined",
                },
                "ref": null,
                "rendered": "undefined, undefined",
                "type": "h2",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <h3 />
              Likes
            </div>,
            <div
              className="follow-divider"
            />,
            <div>
              <h3 />
              Following
            </div>,
            <div
              className="follow-divider"
            />,
            <div>
              <h3 />
              Followers
            </div>,
            <button
              className="follow-button"
              onClick={[MockFunction]}
              type="button"
            >
              FOLLOW
            </button>,
          ],
          "className": "user-follow",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3 />,
                "Likes",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h3",
              },
              "Likes",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "follow-divider",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3 />,
                "Following",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h3",
              },
              "Following",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "follow-divider",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3 />,
                "Followers",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h3",
              },
              "Followers",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "FOLLOW",
              "className": "follow-button",
              "onClick": [MockFunction],
              "type": "button",
            },
            "ref": null,
            "rendered": "FOLLOW",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ariaHideApp": true,
          "bodyOpenClassName": "ReactModal__Body--open",
          "children": Array [
            <i
              className="material-icons close-button"
              onClick={[MockFunction]}
            >
              close
            </i>,
            <h3
              className="modal__title"
            >
              Copy link below to share:
            </h3>,
            "http://localhost/",
          ],
          "className": "modal",
          "closeTimeoutMS": 0,
          "isOpen": false,
          "onRequestClose": [MockFunction],
          "parentSelector": [Function],
          "portalClassName": "ReactModalPortal",
          "role": "dialog",
          "shouldCloseOnEsc": true,
          "shouldCloseOnOverlayClick": true,
          "shouldFocusAfterRender": true,
          "shouldReturnFocusAfterClose": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "close",
              "className": "material-icons close-button",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "close",
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Copy link below to share:",
              "className": "modal__title",
            },
            "ref": null,
            "rendered": "Copy link below to share:",
            "type": "h3",
          },
          "http://localhost/",
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="user-info-container"
          >
            <img
              className="user-img"
              src="user.png"
            />
            <i
              className="material-icons share-icon"
              onClick={[MockFunction]}
            >
              exit_to_app
            </i>
            <div
              className="user-info"
            >
              <h1>
                <i
                  alt="Like"
                  className="material-icons like-icon"
                  onClick={[MockFunction]}
                >
                  favorite_border
                </i>
              </h1>
              <h2>
                undefined, undefined
              </h2>
            </div>
          </div>,
          <div
            className="user-follow"
          >
            <div>
              <h3 />
              Likes
            </div>
            <div
              className="follow-divider"
            />
            <div>
              <h3 />
              Following
            </div>
            <div
              className="follow-divider"
            />
            <div>
              <h3 />
              Followers
            </div>
            <button
              className="follow-button"
              onClick={[MockFunction]}
              type="button"
            >
              FOLLOW
            </button>
          </div>,
          <Modal
            ariaHideApp={true}
            bodyOpenClassName="ReactModal__Body--open"
            className="modal"
            closeTimeoutMS={0}
            isOpen={false}
            onRequestClose={[MockFunction]}
            parentSelector={[Function]}
            portalClassName="ReactModalPortal"
            role="dialog"
            shouldCloseOnEsc={true}
            shouldCloseOnOverlayClick={true}
            shouldFocusAfterRender={true}
            shouldReturnFocusAfterClose={true}
          >
            <i
              className="material-icons close-button"
              onClick={[MockFunction]}
            >
              close
            </i>
            <h3
              className="modal__title"
            >
              Copy link below to share:
            </h3>
            http://localhost/
          </Modal>,
        ],
        "className": "user-data",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                className="user-img"
                src="user.png"
              />,
              <i
                className="material-icons share-icon"
                onClick={[MockFunction]}
              >
                exit_to_app
              </i>,
              <div
                className="user-info"
              >
                <h1>
                  <i
                    alt="Like"
                    className="material-icons like-icon"
                    onClick={[MockFunction]}
                  >
                    favorite_border
                  </i>
                </h1>
                <h2>
                  undefined, undefined
                </h2>
              </div>,
            ],
            "className": "user-info-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": undefined,
                "className": "user-img",
                "src": "user.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "exit_to_app",
                "className": "material-icons share-icon",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": "exit_to_app",
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    <i
                      alt="Like"
                      className="material-icons like-icon"
                      onClick={[MockFunction]}
                    >
                      favorite_border
                    </i>
                  </h1>,
                  <h2>
                    undefined, undefined
                  </h2>,
                ],
                "className": "user-info",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      undefined,
                      <i
                        alt="Like"
                        className="material-icons like-icon"
                        onClick={[MockFunction]}
                      >
                        favorite_border
                      </i>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "Like",
                        "children": "favorite_border",
                        "className": "material-icons like-icon",
                        "onClick": [MockFunction],
                      },
                      "ref": null,
                      "rendered": "favorite_border",
                      "type": "i",
                    },
                  ],
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "undefined, undefined",
                  },
                  "ref": null,
                  "rendered": "undefined, undefined",
                  "type": "h2",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <h3 />
                Likes
              </div>,
              <div
                className="follow-divider"
              />,
              <div>
                <h3 />
                Following
              </div>,
              <div
                className="follow-divider"
              />,
              <div>
                <h3 />
                Followers
              </div>,
              <button
                className="follow-button"
                onClick={[MockFunction]}
                type="button"
              >
                FOLLOW
              </button>,
            ],
            "className": "user-follow",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3 />,
                  "Likes",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h3",
                },
                "Likes",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "follow-divider",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3 />,
                  "Following",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h3",
                },
                "Following",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "follow-divider",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3 />,
                  "Followers",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h3",
                },
                "Followers",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "FOLLOW",
                "className": "follow-button",
                "onClick": [MockFunction],
                "type": "button",
              },
              "ref": null,
              "rendered": "FOLLOW",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "ariaHideApp": true,
            "bodyOpenClassName": "ReactModal__Body--open",
            "children": Array [
              <i
                className="material-icons close-button"
                onClick={[MockFunction]}
              >
                close
              </i>,
              <h3
                className="modal__title"
              >
                Copy link below to share:
              </h3>,
              "http://localhost/",
            ],
            "className": "modal",
            "closeTimeoutMS": 0,
            "isOpen": false,
            "onRequestClose": [MockFunction],
            "parentSelector": [Function],
            "portalClassName": "ReactModalPortal",
            "role": "dialog",
            "shouldCloseOnEsc": true,
            "shouldCloseOnOverlayClick": true,
            "shouldFocusAfterRender": true,
            "shouldReturnFocusAfterClose": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "close",
                "className": "material-icons close-button",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": "close",
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Copy link below to share:",
                "className": "modal__title",
              },
              "ref": null,
              "rendered": "Copy link below to share:",
              "type": "h3",
            },
            "http://localhost/",
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
